src/common/plugin/types.ts

```diff
@@ -83,3 +84,8 @@ export interface LocalWalletMetadata {
   balance: string
   lastSeenBlockHeight: number
 }
+
+export const asLocalWalletMetadataCleaner = asObject({
+  balance: asString,
+  lastSeenBlockHeight: asNumber
+})
```

Comment:
As mentioned, we want to either include `LocalWalletMetadata` static type in the cleaner's definition, or we want to derive the static type from the cleaner in order to maintain code coupling:

Option 1:
```ts
export const asLocalWalletMetadata = asObject<LocalWalletMetadata>({
  balance: asString,
  lastSeenBlockHeight: asNumber
})
```

Option 2 (Preferred/DRY):
```ts
export type LocalWalletMetadata = ReturnType<typeof asLocalWalletMetadata>
export const asLocalWalletMetadata = asObject({
  balance: asString,
  lastSeenBlockHeight: asNumber
})
```

This rule should be applied to all cleaner definitions in this PR.

Reply:
This is nice, will apply everywhere!

Reply:
Done in the newest batch of fixup commits.